From df20ae1280a2601a4c233ec12372d1e17db21e96 Mon Sep 17 00:00:00 2001
From: Thomas Petazzoni <thomas.petazzoni@free-electrons.com>
Date: Mon, 25 Nov 2013 22:51:53 +0100
Subject: [PATCH] Makefile.flags: strip non -l arguments returned by pkg-config

Signed-off-by: Thomas Petazzoni <thomas.petazzoni@free-electrons.com>
[yann.morin.1998@free.fr: refresh for 1.29.0]
[petr.vorel@gmail.com: refresh for 1.32.0]
Signed-off-by: "Yann E. MORIN" <yann.morin.1998@free.fr>
Signed-off-by: Petr Vorel <petr.vorel@gmail.com>
---
 Makefile.flags | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/Makefile.flags b/Makefile.flags
index 97cb4dca2..92a9063d9 100644
--- a/Makefile.flags
+++ b/Makefile.flags
@@ -181,7 +181,9 @@ ifeq ($(CONFIG_SELINUX),y)
 SELINUX_PC_MODULES = libselinux libsepol
 $(eval $(call pkg_check_modules,SELINUX,$(SELINUX_PC_MODULES)))
 CPPFLAGS += $(SELINUX_CFLAGS)
-LDLIBS += $(if $(SELINUX_LIBS),$(SELINUX_LIBS:-l%=%),$(SELINUX_PC_MODULES:lib%=%))
+LDLIBS += $(if $(SELINUX_LIBS),\
+       $(patsubst -l%,%,$(filter -l%,$(SELINUX_LIBS))),\
+       $(SELINUX_PC_MODULES:lib%=%))
 endif
 
 ifeq ($(CONFIG_FEATURE_NSLOOKUP_BIG),y)
-- 
2.48.1

